frontend/src/libs/components/input/input.tsx

```diff
@@ -23,17 +28,65 @@ const Input = <T extends FieldValues>({
   name,
   placeholder = '',
   type = 'text',
+  isActive,
+  isFilled,
+  isDisabled,
 }: Properties<T>): JSX.Element => {
   const { field } = useFormController({ name, control });
 
   const error = errors[name]?.message;
   const hasError = Boolean(error);
+  const mapStatuses = (): string => {
+    let classNames = '';
+
+    if (isActive) {
+      classNames += ' ' + styles.active;
+    }
+
+    if (isFilled) {
+      classNames += ' ' + styles.filled;
+    }
+
+    if (isDisabled) {
+      classNames += ' ' + styles.disabled;
+    }
+
+    if (error) {
+      classNames += ' ' + styles.error;
+    }
+
+    return classNames;
+  };
 
   return (
-    <label>
+    <label className={styles.label}>
       <span>{label}</span>
-      <input {...field} type={type} placeholder={placeholder} />
-      {hasError && <span>{error as string}</span>}
+      {type !== 'textarea' && (
+        <input
+          {...field}
+          type={type}
+          placeholder={placeholder}
+          className={`${styles.input} ${mapStatuses()}`}
+          disabled={isDisabled}
+        />
+      )}
+
+      {type === 'textarea' && (
+        <textarea
+          {...field}
+          placeholder={placeholder}
+          className={`${styles.input} ${mapStatuses()} ${styles.textarea} `}
+          disabled={isDisabled}
+        />
+      )}
+
+      {type === 'password' && (
+        <span className={styles.passwordEye}>&#128065;</span>
+      )}
+
+      <span className={`${styles.errorMessage} ${hasError ? '' : '.hidden'}`}>
+        {error as string}
```

Comment:
which type can be error here besides string? if undefined, so this block should not render at all